presto-hive/src/test/java/com/facebook/presto/hive/TestHiveTypeWidening.java

```diff
@@ -0,0 +1,282 @@
+        TEST_TABLES.put("real_to_real", null);
+        TEST_TABLES.put("real_to_double", null);
+        TEST_TABLES.put("double_to_integer", "The column field of table type_widening_schema\\.double_to_integer is declared as type int, but the Parquet file (.*) declares the column as type DOUBLE");
+        TEST_TABLES.put("double_to_bigint", "The column field of table type_widening_schema\\.double_to_bigint is declared as type bigint, but the Parquet file (.*) declares the column as type DOUBLE");
+        TEST_TABLES.put("double_to_real", "The column field of table type_widening_schema\\.double_to_real is declared as type float, but the Parquet file (.*) declares the column as type DOUBLE");
+        TEST_TABLES.put("double_to_double", null);
+    }
+
+    private DistributedQueryRunner queryRunner;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        logger.info("Creating 'QueryRunner'");
+        Session session = testSessionBuilder().setCatalog(CATALOG).setSchema(SCHEMA).setTimeZoneKey(TimeZoneKey.UTC_KEY).build();
+        this.queryRunner = DistributedQueryRunner.builder(session).setExtraProperties(ImmutableMap.<String, String>builder().build()).build();
+
+        logger.info("  |-- Installing Plugin: " + CATALOG);
+        this.queryRunner.installPlugin(new HivePlugin(CATALOG));
+        Path catalogDirectory = this.queryRunner.getCoordinator().getDataDirectory().resolve("hive_data").getParent().resolve("catalog");
+        logger.info("  |-- Obtained catalog directory: " + catalogDirectory.toFile().toURI());
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put("hive.metastore", "file")
+                .put("hive.metastore.catalog.dir", catalogDirectory.toFile().toURI().toString())
+                .put("hive.allow-drop-table", "true")
+                .put("hive.non-managed-table-writes-enabled", "true")
+                .put("hive.parquet.use-column-names", "true")
+                .put("hive.compression-codec", "GZIP")
+                .put("hive.storage-format", "PARQUET")
+                .build();
+        logger.info("  |-- Properties loaded");
+
+        logger.info("  |-- Creating catalog '" + CATALOG + "' using plugin '" + CATALOG + '\'');
+        this.queryRunner.createCatalog(CATALOG, CATALOG, properties);
+        logger.info("  |-- Catalog '" + CATALOG + "' created");
+        logger.info("  |-- Creating schema '" + SCHEMA + "' on catalog '" + CATALOG + '\'');
+        this.queryRunner.execute(format("CREATE SCHEMA %s.%s", CATALOG, SCHEMA));
+        logger.info("  |-- Schema '" + SCHEMA + "' created");
+
+        logger.info("'QueryRunner' created succesfully");
+        return this.queryRunner;
+    }
+
+    @DataProvider
+    public Object[][] dataProvider()
```

Comment:
While using a data provider seems to reduce the LOCs for this file quite a bit, I am finding it hard to grok.  I think if we just don't use the data provider and create utilities for repetitive operations, it might make the test file larger, but more straightforward to understand.  Can you give it a try?

Reply:
Refactor done:
Removed usage of DataProvider. Replaced test method with multiple tests, one for each use case. The table creation, deletion and query logic commonly used by tests is centralized in several utility methods.